R/USA.R

```diff
@@ -0,0 +1,126 @@
+          "US-MI", "US-MN", "US-MS", "US-MO", "US-MT", "US-NE", "US-NV",
+          "US-NH", "US-NJ", "US-NM", "US-NY", "US-NC", "US-ND", "US-OH",
+          "US-OK", "US-OR", "US-PA", "US-RI", "US-SC", "US-SD", "US-TN",
+          "US-TX", "US-UT", "US-VE", "US-VA", "US-WA", "US-WV", "US-WI",
+          "US-WY", "US-DC", "US-AS", "US-GU", "US-MP", "US-PR", "US-UM",
+          "US-VI"
+        ),
+        region = c(
+          "Alabama", "Alaska", "Arizona", "Arkansas", "California", "Colorado",
+          "Connecticut", "Delaware", "Florida", "Georgia", "Hawaii", "Idaho",
+          "Illinois", "Indiana", "Iowa", "Kansas", "Kentucky", "Louisiana",
+          "Maine", "Maryland", "Massachusetts", "Michigan", "Minnesota",
+          "Mississippi", "Missouri", "Montana", "Nebraska", "Nevada",
+          "New Hampshire", "New Jersey", "New Mexico", "New York",
+          "North Carolina", "North Dakota", "Ohio", "Oklahoma", "Oregon",
+          "Pennsylvania", "Rhode Island", "South Carolina", "South Dakota",
+          "Tennessee", "Texas", "Utah", "Vermont", "Virginia", "Washington",
+          "West Virginia", "Wisconsin", "Wyoming", "District of Columbia",
+          "American Samoa", "Guam", "Northern Mariana Islands", "Puerto Rico",
+          "Minor Outlying Islands", "Virgin Islands"
+        )
+      )
+      self$codes_lookup$`2` <- self$codes_lookup$`1`
+    },
+
+    #' @description USA specific state level data cleaning
+    #'
+    clean_common = function() {
+      if (self$level == "1") {
+        self$clean_level_1()
+      } else if (self$level == "2") {
+        self$clean_level_2()
+      }
+    },
+
+    #' @description USA Specific State Level Data Cleaning
+    #' @importFrom dplyr rename mutate select left_join
+    clean_level_1 = function() {
+      self$data$clean <- self$data$raw$`1` %>%
+        rename(
+          level_1_region = state,
+          cases_total = cases,
+          deaths_total = deaths
+        ) %>%
+        mutate(
+          cases_total = replace(cases_total, cases_total < 0, 0),
+          deaths_total = replace(deaths_total, deaths_total < 0, 0)
+        ) %>%
+        select(date, level_1_region, cases_total, deaths_total) %>%
+        left_join(self$codes_lookup[[1]],
+          by = c("level_1_region" = "region"),
+          copy = TRUE
+        )
+    },
+
+    #' @description USA Specific County Level Data Cleaning
+    #' @importFrom dplyr mutate rename left_join
+    clean_level_2 = function() {
+      self$data$clean <- self$data$raw$`2` %>%
+        rename(
+          level_2_region = county,
+          level_1_region = state,
+          level_2_region_code = fips,
+          cases_total = cases,
+          deaths_total = deaths
+        ) %>%
+        mutate(
+          cases_total = replace(cases_total, cases_total < 0, 0),
+          deaths_total = replace(deaths_total, deaths_total < 0, 0)
+        ) %>%
+        left_join(self$codes_lookup[[2]],
```

Comment:
If these are really just level 1 regions we don't need region codes duplicated that are exactly the same. Should just be able to either link the lookup table in clean_common or if need cleaning variables just do it afterwards using the level 1 lookup table. 

If these codes are the same why are they defined as iso codes and as fips codes ( @kathsherratt ?)

Reply:
What was the logic behind this?

Reply:
Fips is included with the level 2 data and is a numeric code (e.g. 01001, 01002, ...)